basculechecks/basculechecks.go

```diff
@@ -14,18 +13,11 @@ var (
 	ErrNoVals                 = errors.New("expected at least one value")
 	ErrNoAuth                 = errors.New("couldn't get request info: authorization not found")
 	ErrNonstringVal           = errors.New("expected value to be a string")
-	ErrEmptyString            = errors.New("expected string to be nonempty")
 	ErrNoValidCapabilityFound = errors.New("no valid capability for endpoint")
 )
 
-type CapabilityConfig struct {
-	FirstPiece      string
-	SecondPiece     string
-	ThirdPiece      string
-	AcceptAllMethod string
-}
-
-func CreateValidCapabilityCheck(config CapabilityConfig) func(context.Context, []interface{}) error {
+func CreateValidCapabilityCheck(prefix string, acceptAllMethod string) func(context.Context, []interface{}) error {
+	matchPrefix := regexp.MustCompile("^" + prefix + "(.+):(.+?)$")
```

Comment:
this should be error checked,  or some modification/validation should happen to `prefix`